File: subroutine-calling-conventions/coeficiente_binomial.py
# Calcula el coeficiente binomial de n en k
def coeficiente_binomial(n, k):
    if k == 0 or k == n:
        return 1
    else:
        return coeficiente_binomial(n - 1, k - 1) + coeficiente_binomial(n - 1, k)

def main():
    n = int(input("Introduce el valor de n: "))
    k = int(input("Introduce el valor de k: "))
    print(coeficiente_binomial(n, k))

if __name__ == '__main__':
    main()

File: subroutine-calling-conventions/CoeficienteBinomial.cpp
// Calcula el coeficiente binomial de n en k

#include <bits/stdc++.h>

using namespace std;

int coeficienteBinomial(int n, int k) {
    if (k == 0 || k == n) {
        return 1;
    }
    return coeficienteBinomial(n - 1, k - 1) + coeficienteBinomial(n - 1, k);
}

int main() {
    ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);

    int n, k;
    cout << "Introduce n: ";
    cin >> n;
    cout << "Introduce k: ";
    cin >> k;

    cout << "El coeficiente binomial es: " << coeficienteBinomial(n, k) << endl;

    return 0;
}